kore/src/Kore/Step/Rule.hs

```diff
@@ -777,3 +780,48 @@ substitute subst rulePattern' =
         }
   where
     RulePattern { left, right, requires, ensures } = rulePattern'
+
+{- | Context in which a step is taken
+-}
+data StepContext
+  = SimplificationContext
+  | DefinitionContext
+  | RewriteContext
+  | OnePathContext
+  | AllPathContext
+  deriving (Show, Eq)
+
+
+{- | Check if the term and the 'left' fields of the rules are function-like
+
+-}
+
+checkFunctionLike
+    ::  (Debug variable
+        , FreshVariable variable
+        , SortedVariable variable
+        , Show variable
+        , Unparse variable
+        , Debug variable'
+        , FreshVariable variable'
+        , SortedVariable variable'
+        , Show variable'
+        , Unparse variable')
+    => [RulePattern variable']
+    -> TermLike variable
+    -> Either String ()
+checkFunctionLike [] _ = pure ()
+checkFunctionLike rules term
+  | TermLike.isFunctionPattern term =
```

Comment:
Why is it OK to omit this check if we did not apply any rules? There would still be a remainder pattern, right?

Reply:
I think that if we don't apply any rules then there is no change to the original pattern, so, in principle, the remainder should be itself.